### ISSUE TITLE
[SDK 50] Shared Element Transitions stopped working

### ISSUE THREAD

**Issue by TMaszko** [expo/expo#27606]
### Minimal reproducible example

https://snack.expo.dev/@tmaszko/sharedtransitiontest

### What platform(s) does this occur on?

iOS

### Did you reproduce this issue in a development build?

Yes

### Summary

Expected: Shared element transition works as it suppose to.
It works with the bare react native app .

### Environment

System:
      OS: macOS 14.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.19.0 - ~/.nvm/versions/node/v18.19.0/bin/node
      Yarn: 1.22.21 - ~/.nvm/versions/node/v18.19.0/bin/yarn
      npm: 10.2.3 - ~/.nvm/versions/node/v18.19.0/bin/npm
      Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.4, iOS 17.4, macOS 14.4, tvOS 17.4, visionOS 1.1, watchOS 10.4
    IDEs:
      Android Studio: 2023.2 AI-232.10227.8.2321.11479570
      Xcode: 15.3/15E204a - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.11 => 50.0.11 
      react: 18.2.0 => 18.2.0 
      react-native: 0.73.4 => 0.73.4 
    Expo Workflow: managed

### Expo Doctor Diagnostics

✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that native modules use compatible support package versions for installed Expo SDK
✔ Check that packages match versions required by installed Expo SDK

**Comment by brentvatne** [comment]
hi there! are you saying that something that works outside of expo go is not working inside of expo go?

**Comment by TMaszko** [comment]
Hi @brentvatne!  Exactly :)
I've installed all packages so versions are matching and bare react native project works just fine

**Comment by brentvatne** [comment]
@TMaszko - if you run `npx expo run:ios` in the project where you have this issue in expo go, so that it runs "prebuild" and then compiles the runs the project, are you able to reproduce it?

**Comment by TMaszko** [comment]
No I'm not :( It's working just fine as in the bare react native app

**Comment by brentvatne** [comment]
@TMaszko - got it! I tried `npx expo run:ios` locally and indeed it worked correctly, so it looks like this is an issue with how reanimated is vendored in Expo Go

**Comment by junioroga** [comment]
eagerly waiting for this fix.

**Comment by tsalama** [comment]
Likewise.

**Comment by BitPhinix** [comment]
Experiencing this as well

**Comment by shantanu-daisy** [comment]
mine does not work with dev builds at all. also getting a warning "[Reanimated] Duplicate shared tag "sharedFooter" on the same screen" but in the reanimated repo, seems like people are having issue with expo 50 but not 49.

**Comment by killah-t-cell** [comment]
> mine does not work with dev builds at all. also getting a warning "[Reanimated] Duplicate shared tag "sharedFooter" on the same screen" but in the reanimated repo, seems like people are having issue with expo 50 but not 49.

Does it still work in prod though?

**Comment by codegabk** [comment]
We need a fix please :(

**Comment by ABertaud** [comment]
it's looking like with SDK 50 and `expo-router` shared element transition aren't working at all.
when i'm migrating expo-router to react-navigation, I get the same laggy behaviour as mentioned in the issue.

note that for SDK 49 whether it is with `expo-router` or `react-navigation,` shared element transition goes well.

**Comment by cdoucy** [comment]
Hello @brentvatne , is it fix in SDK 51?
Thanks!

**Comment by brentvatne** [comment]
@cdoucy - give it a try and let me know! either way, this only impacts expo go - you should use development builds in most cases

**Comment by ABertaud** [comment]
Hello @brentvatne, this bug is still persistent when using `expo-router` over `react-navigation`.
It's not longer possible to make shared element transition with `expo-router` from sdk 50, have you investigated it?

**Comment by brentvatne** [comment]
@ABertaud - as far as i know this only impacts expo go. are you saying it impacts development and release builds too?

**Comment by ABertaud** [comment]
@brentvatne - yes, it impacts development and release builds but only when using `expo-router` instead of `react-navigation`. which is quite annoying because having to re-migrate everything to react-nav... takes time!

**Comment by brentvatne** [comment]
@ABertaud - can you create an issue with a minimal reproducible example? this issue is about expo go, and as others have said in this issue, it is working fine for them in dev builds.

**Comment by ABertaud** [comment]
> "useFrameworks": "static"

@siderakis I'm not using it.

**Comment by gowri-xflowpay** [comment]
Having same issue

**Comment by codegabk** [comment]
The issue is still there on expo 51 for expo go 😕

**Comment by brentvatne** [comment]
this isn't a priority for us to resolve in expo go given that it is just a sandbox environment, please use a development build if this is important for you. we will revisit this in the future when we have time, but [shared element transitions are an experimental feature](https://docs.swmansion.com/react-native-reanimated/docs/shared-element-transitions/overview/) and you may encounter issues like this

**Comment by ioeldev** [comment]
I want to add that this issue also happens with development build, not only expo-go. When using expo-router, i expect to have the same features that react-navigation and having this shared element transition is part of it. Is it possible to understand what differs ? We don't want to move out from expo-router in our project :)

**Comment by RaulFaife** [comment]
Just going to upvote on here too. Been finding people in YT tutorials and stackoverflow asking about this not knowing this thread exists. Would definitely be super useful!

**Comment by KimGnab** [comment]
I'm still having issues with the expo 51 dev build.

**Comment by lourxd** [comment]
Issue still persistent
I am using development build with
- reanimated 3.10.1
- react native screens: 3.31.1
- @react-navigation/native-stack: 6.11.0
- expo: 51.0.37
- react-native : 0.74.5

First time I enter the stack navigator and I click on the element to navigate on the other page of the stack everything works good, then I go into my drawer and I unmount this stack, each time I try to mount this stack again it freezes the app and I have to reboot it.
I tried making unmountOnBlur: true and unmountOnBlur: false for the drawer navigator, but is crashes when unmountOnBlur: false.

**Comment by POLEC4T** [comment]
I'm having this issue with a dev. build made with eas

**Comment by harrigee** [comment]
We are also facing the same issue

**Comment by buildnewapp** [comment]
@joekendal Can you explain in detail? Or provide a simple demo, thank you.

**Comment by ElJosecito** [comment]
There is any update in this issue?

I really want to make this behavior in my app and i'm using expo 52 realesed version, with the file based routing.

**Comment by ElJosecito** [comment]
@joekendal Hey can you provide a demo please, I've tried your folder layout solution but it didn't work, thank you.

**Comment by philscott-dev** [comment]
@joekendal stumbled into this thread just now. I'm having this issue as well on 52, and I've set `"newArchEnabled": false`. Any guidance on what you did to address this?

**Comment by philscott-dev** [comment]
Yep I've rebuild and I've tried in Expo Go and an iOS development build. no luck. If I get some time I'll whip up a minimal repo and try again.

**Comment by kouloughli-hemza** [comment]
im not sure if this is even supported using expo router v3, always not working!

**Comment by marcopolodesign** [comment]
Any updates here?

**Comment by chetwanimanish** [comment]
Tried switching to old architecture now the animation of shared element works, but, just once in iOS and then the app crashes. In Android it is working perfectly fine.

**Comment by TerryMinn** [comment]
i think that is not working in new arch b/c i am switch to old arch and that was correctly work i don't know why

**Comment by kitten** [comment]
cc @brentvatne: I see we accepted this, but this isn't something we can address in Expo Router, I suppose? We should maybe close this as a downstream issue, right?

**Comment by brentvatne** [comment]
@kitten - this is an issue that spans reanimated and expo go, but ultimately the reanimated team is in the process of rewriting shared element transitions for reanimated v4 anyhow. reanimated shared element transitions do not support the new architecture, so that is why they are not working in expo go currently. when the reanimated team adds support for SETs with the new architecture we will work with the team to ensure it works well in expo go. in the meantime, i'll lock this thread as it's a known issue.

shared element transitions may work for you in reanimated in a development / release build, but keep in mind the warning from the reanimated docs:

<img width="922" height="249" alt="Image" src="https://github.com/user-attachments/assets/c908076e-55a4-46a7-a795-a0284dfc21b4" />
